heart/heartbeat.py

```diff
@@ -234,10 +258,7 @@ def collect(self, ct):
 			self.parse_http_data(data)
 	
 	def push_buf(self, ct):
-		if self.sockfd == None:
-			return
-
-		if self.wbuf == None:
+		if None in (self.sockfd, self.wbuf):
```

Comment:
This looks nasty. I'd have just done `if self.sockfd == None or self.wbuf == None:`.